pymeasure/display/windows/dock_window.py

```diff
@@ -0,0 +1,200 @@
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+
+from pyqtgraph.dockarea import Dock, DockArea
+
+from .managed_window import ManagedWindowBase
+from ..Qt import QtCore, QtWidgets
+from ..widgets import (
+    PlotWidget,
+    LogWidget
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class DockWindow(ManagedWindowBase):
+    """
+    Display experiment output with an :class:`~pymeasure.display.widgets.image_widget.ImageWidget`
+    class.
+
+    :param procedure_class: procedure class describing the experiment (see
+        :class:`~pymeasure.experiment.procedure.Procedure`)
+    :param x_axis: the data column(s) for the x-axis of the plot. This may be string or a list
+        of strings from the data columns of the procedure.
+    :param y_axis: the data column(s) for the y-axis of the plot. This may be string or a list
+        of strings from the data columns of the procedure.
+    :param num_plots: the number of plots you want displayed in the DockWindow tab
+    :param \\**kwargs: optional keyword arguments that will be passed to
+        :class:`~pymeasure.display.windows.managed_window.ManagedWindowBase`
+    """
+
+    def __init__(self, procedure_class, x_axis=None, y_axis=None, num_plots=1, *args, **kwargs):
+
+        self.x_axis = x_axis
+        self.y_axis = y_axis
+        self.num_plots = num_plots
+
+        self.log_widget = LogWidget("Experiment Log")
+        self.widget_list = []
+
+        super().__init__(
+            procedure_class=procedure_class,
+            setup=False,
+            widget_list=self.widget_list,
+            *args,
+            **kwargs
+        )
+
+        self._setup_ui()
+        self._layout()
+
+        measure_quantities = []
+        if type(self.x_axis) == list:
+            # Expand x_axis if it is a list
+            measure_quantities += [*self.x_axis]
+        else:
+            measure_quantities.append(self.x_axis)
+        if type(self.y_axis) == list:
+            # Expand y_axis if it is a list
+            measure_quantities += [*self.y_axis]
+        else:
+            measure_quantities.append(self.y_axis)
+
+        self.browser_widget.browser.measured_quantities = measure_quantities
+
+        logging.getLogger().addHandler(self.log_widget.handler)  # needs to be in Qt context?
+        log.setLevel(self.log_level)
+        log.info("DockWindow connected to logging")
+
+    def _layout(self):
```

Comment:
I don't think it is necessary to redefine the `_layout` method. See my general comment.

Reply:
In last refactor I moved this logic to DockWidget